src/test/resources/com/puppycrawl/tools/checkstyle/checks/metrics/classfanoutcomplexity/InputClassFanOutComplexityAnnotations.java

```diff
@@ -0,0 +1,26 @@
+package com.puppycrawl.tools.checkstyle.checks.metrics.classfanoutcomplexity;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+public class InputClassFanOutComplexityAnnotations {
+
+    public void foo1(@TypeAnnotation char a){}
+
+    public void foo2(final char @TypeAnnotation [] a){}
+
+    @MethodAnnotation
+    public void foo3(){}
+
+    @Override
+    public String toString() {
+        return super.toString();
+    }
+
+}
+
+@Target(ElementType.TYPE_USE)
```

Comment:
I think both `@Target` and `ElementType` should be counted

Reply:
Do you think need counted for `TypeAnnotation`?

Reply:
yes

Reply:
fixed. But `Target` and` Element` are ignored, because is contained `java.lang.*`